### PR TITLE
18.11 hw16

### PR DESCRIPTION
**Pull request by Dergaeva**
Correct all issue. Please check

### CODE REVIEW

**Comment by iliatcymbal** [comment]
[http://joxi.ru/eAO1JlgSxzvkDm](http://joxi.ru/eAO1JlgSxzvkDm)
There are warnings from react, you should fix them

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iliatcymbal) [inline .idea/codeStyleSettings.xml]: Why you have .idea files in repository?
Everytime I see this unnecessary history

File: src/footer/Footer.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './footer.scss';
+
+import { Username } from './userList';
+import { users } from './userList';
+import { ListComponentName } from './userList';
+
+export const Footer = () => (
+  <footer className="footer">
+    <Username fName='Dergaieva'  lName='Elena' age='30' />
```
Comment (iliatcymbal) [inline src/footer/Footer.js]: UserName should be inside ListComponentName

File: src/footer/userList/userList.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import './userList.scss';
+
+export const users =  [
```
Comment (iliatcymbal) [inline src/footer/userList/userList.js]: this list shouldn't be here. You can save it in separate file, or define this array in Footer.js

File: src/footer/userList/userList.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <span>{props.age} </span>
+    </div>
+  );
+};
+
+export const ListComponentName = (props) => {
+  return (
+    <div class="name">
+    <ul>{
+      props.usersList.map(user => <li key={user.id}>{user.fName} {user.lName} {user.age}</li>)
```
Comment (iliatcymbal) [inline src/footer/userList/userList.js]: You should use here Username component
```
<li key={user.id}><Username ... /></li>
````

File: src/header/number/Number.js [hunk 5]
```diff
@@ -3,48 +3,23 @@ import './number.scss';
...
-					))
-				}
-			</ul>
-		)
+    arr = arr.filter(n => n % 2 === 0);
 	}
-	else
-		count();
-		return (
+	return (
```
Comment (iliatcymbal) [inline src/header/number/Number.js:17]: empty line before return

File: src/header/welcome/Welcome.js [hunk 6]
```diff
@@ -4,19 +4,22 @@ import './welcome.scss';
 
 export const Welcome = ({ name }) => {
 	var hr = (new Date()).getHours();
+	let text;
 	if(hr >= 22 || hr <= 3) {
```
Comment (iliatcymbal) [inline src/header/welcome/Welcome.js]: space after if

File: src/header/welcome/Welcome.js [hunk 7]
```diff
@@ -4,19 +4,22 @@ import './welcome.scss';
...
-		return <h2>Good night, {name}!</h2>
+		text = 'Good night';
 	}
 	if(hr > 3 || hr <= 12) {
```
Comment (iliatcymbal) [inline src/header/welcome/Welcome.js]: space after if

File: src/header/welcome/Welcome.js [hunk 8]
```diff
@@ -4,19 +4,22 @@ import './welcome.scss';
...
-		return <h2>Good morning, {name}!</h2>
+		text = 'Good morning';
 	}
 	if(hr > 12 || hr <= 18) {
```
Comment (iliatcymbal) [inline src/header/welcome/Welcome.js]: space after if

File: src/header/welcome/Welcome.js [hunk 9]
```diff
@@ -4,19 +4,22 @@ import './welcome.scss';
...
-		return <h2>Good afternoon, {name}!</h2>
+		text = 'Good afternoon';
 	}
 	if(hr > 18 || hr < 22) {
-		return <h2>Good evening, {name}!</h2>
+		text = 'Good evening';
 	}
 	else
-		return <h2>What time is it, {name}?</h2>
+		text = 'What time is it';
```
Comment (iliatcymbal) [inline src/header/welcome/Welcome.js]: curly braces after else

**Review by iliatcymbal — COMMENTED** [review]
Please, fix all comments

File: src/footer/userList/userList.js [hunk 10]
```diff
@@ -35,8 +14,9 @@ export const Username = (props) => {
...
   return (
-    <div class="name">
-    <ul>{
-      props.usersList.map(user => <li key={user.id}>{user.fName} {user.lName} {user.age}</li>)
-    }</ul></div>
-  )};
+    <div className="name">
+      <Username fName='Dergaieva'  lName='Elena' age='30' />
+      <ul>{
+        props.usersList.map(user => <li key={user.id}>{user.fName} {user.lName} {user.age}</li>)
```
Comment (iliatcymbal) [inline src/footer/userList/userList.js]:
```
props.usersList.map(user => <li key={user.id}>{user.fName} {user.lName} {user.age}</li>)
```
Inside < li>< /li> shoul be placed a Username component
there  was my [comment](https://github.com/Dergaeva/dergaeva.es-trainapp/pull/3#discussion_r154475697)
It is still not fixed

**Comment by Dergaeva** [comment]
@iliatcymbal 
`export const ListComponentName = (props) => {
  return (
    <div className="name">
      <ul>{
        props.usersList.map(user => <li key={user.id}><Username fName={user.fName}  lName={user.lName} age={user.age} /></li>)
      }</ul>
    </div>
)};`

**Comment by iliatcymbal** [comment]
Good, you can merge to master

**Pull request merged**